src/hamiltonian.jl

```diff
@@ -0,0 +1,77 @@
+# TODO: add a type for kinetic energy
+
+struct Hamiltonian{T<:Real,M<:AbstractMetric{T},F1,F2,A<:AbstractVector{T}}
+    metric      ::  M
+    logπ        ::  F1
+    dlogπdθ     ::  F2
+    # Below are for efficient memory allocation
+    _dHdθ       ::  A
+    _dHdr       ::  A
+    _r          ::  A
```

Comment:
Some comment on variable meaning would be helpful.

Reply:
I put a note in `HMC.jl` on the notations and the variable name is actually informative enough on what's the term is. But sure we can improve the documentation a bit if we want.